src/axom/core/utilities/BitUtilities.hpp

```diff
@@ -23,14 +23,14 @@
 
 // Check for and setup defines for platform-specific intrinsics
 // Note: `__GNUC__` is defined for the gnu, clang and intel compilers
-#if defined(AXOM_USE_CUDA)
-  // Intrinsics included implicitly
-#elif defined(_WIN64) && (_MSC_VER >= 1600)
+#if defined(_WIN64) && (_MSC_VER >= 1600)
   #define _AXOM_CORE_USE_INTRINSICS_MSVC
   #include <intrin.h>
 #elif defined(__x86_64__) && defined(__GNUC__)
   #define _AXOM_CORE_USE_INTRINSICS_GCC
-  #include <x86intrin.h>
+  #if !defined(__CUDACC__)
```

Comment:
I'm not convinced this is right, either. I can't reproduce the error of including x86intrin.h.

> nvcc -ccbin=/usr/tce/packages/gcc/gcc-11.2.1-magic/bin/gcc -Xcompiler -mavx512f -c meh.cu

> cat meh.cu
`#include <x86intrin.h>`

Reply:
Another possibility is to get rid of the intrinsics and use a definition that always works, or to define these functions in a source file.

Reply:
@adayton1 what was the error you were seeing?

Reply:
Related issues:
https://github.com/LLNL/axom/issues/798

Related PRs:
https://github.com/LLNL/axom/pull/1088
https://github.com/LLNL/axom/pull/1180

Reply:
PR #1088 was supposedly a fix for issue #798, but that introduced the ODR violation.

Reply:
I can't reproduce #798 with the example mentioned in https://forums.developer.nvidia.com/t/nvcc-with-avx-support-cannot-find-gcc-builtin-intrinsics/35120. So maybe the best thing to do would be to put the code back the way it was before #1088.